metals/src/main/scala/scala/meta/internal/metals/FormattingProvider.scala

```diff
@@ -398,101 +406,120 @@ final class FormattingProvider(
   }
 
   def validateWorkspace(projectRoot: AbsolutePath): Future[Unit] = {
-    if (scalafmtConf(projectRoot).exists) {
-      val text = scalafmtConf(projectRoot).toInputFromBuffers(buffers).text
-      ScalafmtConfig.parse(text) match {
-        case Failure(e) =>
-          scribe.error(s"Failed to parse ${scalafmtConf(projectRoot)}", e)
-          Future.unit
-        case Success(values) =>
-          checkIfDialectUpgradeRequired(values, projectRoot)
-      }
-    } else {
-      Future.unit
+    scalafmtConf(projectRoot) match {
+      case Some(conf) =>
+        val text = conf.toInputFromBuffers(buffers).text
+        ScalafmtConfig.parse(text) match {
+          case Failure(e) =>
+            scribe.error(s"Failed to parse ${conf}", e)
+            Future.unit
+          case Success(values) =>
+            checkIfDialectUpgradeRequired(values, conf)
+        }
+      case None =>
+        Future.unit
     }
   }
 
-  private def scalafmtConf(projectRoot: AbsolutePath): AbsolutePath = {
+  private def scalafmtConf(projectRoot: AbsolutePath): Option[AbsolutePath] = {
     val configpath = userConfig().scalafmtConfigPath
-    configpath.getOrElse(projectRoot.resolve(".scalafmt.conf"))
+    val default: Option[AbsolutePath] = {
+      val defaultLocation = projectRoot.resolve(defaultScalafmtLocation)
+      lazy val scalacliDefault =
+        projectRoot.resolve(".scala-build/.scalafmt.conf")
+      lazy val hiddenDefault = projectRoot.resolve(Directories.hiddenScalafmt)
+      if (defaultLocation.exists) Some(defaultLocation)
+      else if (scalacliDefault.exists) Some(scalacliDefault)
+      else if (hiddenDefault.exists) Some(hiddenDefault)
+      else None
```

Comment:
```suggestion
      List(defaultLocation, scalacliDefault, hiddenDefault).find(_.exists)
```
You can use lazy list, if it's laziness you're after.

Reply:
The laziness was not important, good idea with the list!